library/src/c_api/mod.rs

```diff
@@ -183,6 +194,20 @@ fn path_to_c_string(path: Option<PathBuf>) -> anyhow::Result<*mut c_char> {
     })
 }
 
+fn to_update_result(status: anyhow::Result<UpdateStatus>) -> UpdateResult {
+    let result = match status {
+        Ok(status) => UpdateResult {
+            status: status as i32,
+            message: std::ptr::null(),
+        },
+        Err(err) => UpdateResult {
+            status: SHOREBIRD_UPDATE_ERROR,
```

Comment:
We should probably make this -1 or 100 or something to avoid later colliding with status?